### DISCUSSION TITLE
Pre/Post-request scripts: interpolation of variables with functions

### DISCUSSION THREAD

**Discussion by andreygrigoriev** [Kong/insomnia#7832] [Help] [answered]
Hi,

Is it possible to resolve an environment variable having a function inside? I'm using a plugin `insomnia-plugin-azure-keyvault-secrets` and trying to get a secret value inside a pre-request script, like this (`apiKey` is an environment variable):
```javascript
var apiKey = insomnia.environment.replaceIn("{{apiKey}}");
console.log('apiKey', apiKey);
```
But it shows me a definition of the variable rather than the resolved value:
```
* log: "apiKey"
* "{% azureSecret 'APIKEY' %}"
```
If I use the same variable in the request as a header, it resolves correctly

**Comment by ihexxa** [comment, marked as answer]
Hi,
It is not supported yet, is it possible to let the plugin to save the key as an environment value?
Probably another workaround is accessing the key manually in the script I guess.

**Reply by andreygrigoriev** [reply]
Not sure, that I understood the last one. What do you mean by accessing the key manually in the script?

**Reply by ihexxa** [reply]
It seems that the plugin relies on @azure's module so it would be a bit complicated to manually get apiKey: https://github.com/glconti/insomnia-plugin-azure-keyvault-secrets/blob/main/app.js#L17
My original idea was the apikey could be fetched by send a request in the pre-request script.

**Reply by andreygrigoriev** [reply]
Okay, got it. Thanks. But you say that there is a plan to support functions, right? Can I create somewhere something like a feature request that you can add it to the roadmap?

**Reply by ihexxa** [reply]
I've created a task for it and see if there is a quick solution for it (after the runner release i guess), or it will be prioritized with other tasks in queue.
